pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -441,33 +441,55 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 
 	// If the liveness probe is enabled
 	podTemplateSpec.Spec.Containers[0] = opconfig.ConfigureLivenessProbe(cephv1.KeyOSD, podTemplateSpec.Spec.Containers[0], c.spec.HealthCheck)
-
+	deployment := &apps.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deploymentName,
+			Namespace: c.clusterInfo.Namespace,
+			Labels:    c.getOSDLabels(osd.ID, failureDomainValue, osdProps.portable),
+		},
+	}
 	if c.spec.Network.IsHost() {
 		podTemplateSpec.Spec.DNSPolicy = v1.DNSClusterFirstWithHostNet
 	} else if c.spec.Network.NetworkSpec.IsMultus() {
-		k8sutil.ApplyMultus(c.spec.Network.NetworkSpec, &podTemplateSpec.ObjectMeta)
+		if err := k8sutil.ApplyMultus(c.spec.Network.NetworkSpec, &podTemplateSpec.ObjectMeta); err != nil {
+			return deployment, err
+		}
 	}
 
-	deployment := &apps.Deployment{
+	/*deployment := &apps.Deployment{
```

Comment:
Let's also restore this code and return nil above when there is failure.

Reply:
Done.